### DISCUSSION TITLE
Welcome to fmodel-rust Discussions!

### DISCUSSION THREAD

**Discussion by idugalic** [fraktalio/fmodel-rust#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by RuggeroRebellato** [comment]
Great Work, Ivan!

I just wanted to take a moment to congratulate you on the incredible work you've done with the f(model) library. I've been exploring it for some time, I believe it holds immense promise. Keep up the fantastic work! 👏👏👏.

**Reply by idugalic** [reply]
Thank you so much, Ruggero!

I am happy to hear that it is useful for you. It holds immense promise, indeed!
Rust is a wonderful programming language. It fits `domain modeling` very well. 

For your information, `f(model)` library is implemented in a couple of languages: Kotlin, TypeScript, and Rust (in this repo).
You can read more about it here: [https://fraktalio.com/domain-modeling](https://fraktalio.com/domain-modeling).

Public demos:

#### FModel Demo Applications
|        | Event-Sourced  | State-Stored   |
| :---   |     :---:      |     :---:      |
| `Kotlin` (Spring) |  [fmodel-spring-demo](https://github.com/fraktalio/fmodel-spring-demo) | [fmodel-spring-state-stored-demo](https://github.com/fraktalio/fmodel-spring-state-stored-demo) |
| `Kotlin`(Ktor)   |  [fmodel-ktor-demo](https://github.com/fraktalio/fmodel-ktor-demo)     |    todo     |
| `TypeScript`     |  [fmodel-gift-card-ts](https://github.com/axoniq/giftcard-demo-ts)     |    todo     |
| `Rust`           |  [fmodel-rust-demo](https://github.com/fraktalio/fmodel-rust-demo), [fmodel-gift-card-rust](https://github.com/AxonIQ/axon-rust/tree/main/gift-card-rust)     |    todo     |

Thanks again, and let me know if you have questions/suggestions.

Best,
Ivan
